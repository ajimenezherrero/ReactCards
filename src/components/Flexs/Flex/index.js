import styled from 'styled-components';
import PropTypes from 'prop-types';

const Flex = styled.div`
  ${({ height }) => height && `height: ${height};`};
  width: ${({ width }) => width ? `${width}` : '100%'};
  display: ${({ inline }) => inline ? 'inline-flex' : 'flex'};
  flex-direction: ${({ column }) => column ? 'column' : 'row'};
  flex-wrap: ${({ wrap }) => wrap ? 'wrap' : 'nowrap'};

  ${({ alignEnd }) => alignEnd && 'align-items: flex-end;'}
  ${({ alignStart }) => alignStart && 'align-items: flex-start;'}
  ${({ alignStrech }) => alignStrech && 'align-items: strech;'}
  ${({ alignCenter }) => alignCenter && 'align-items: center;'}
  ${({ alignBaseline }) => alignBaseline && 'align-items: baseline;'}

  ${({ justifyEnd }) => justifyEnd && 'justify-content: flex-end;'}
  ${({ justifyStart }) => justifyStart && 'justify-content: flex-start;'}
  ${({ justifyCenter }) => justifyCenter && 'justify-content: center;'}
  ${({ justifyAround }) => justifyAround && 'justify-content: space-around;'}
  ${({ justifyBetween }) => justifyBetween && 'justify-content: space-between;'}
`;

Flex.propTypes = {
  height: PropTypes.string,
  width: PropTypes.string,
  inline: PropTypes.bool,
  column: PropTypes.bool,
  wrap: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string
  ]),
  alignEnd: PropTypes.bool,
  alignStart: PropTypes.bool,
  alignStrech: PropTypes.bool,
  alignCenter: PropTypes.bool,
  alignBaseline: PropTypes.bool,
  justifyEnd: PropTypes.bool,
  justifyStart: PropTypes.bool,
  justifyCenter: PropTypes.bool,
  justifyAround: PropTypes.bool,
  justifyBetween: PropTypes.bool
};

export default Flex;
