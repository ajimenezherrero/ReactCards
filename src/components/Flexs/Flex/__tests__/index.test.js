import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Flex from '../index';

describe('<Flex />', () => {
  it('renders div with inline flex', () => {
    const tree = renderer.create(
      <Flex inline />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex column direction', () => {
    const tree = renderer.create(
      <Flex column />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex aligned to the end', () => {
    const tree = renderer.create(
      <Flex alignEnd />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex aligned to the start', () => {
    const tree = renderer.create(
      <Flex alignStart />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex aligned strech', () => {
    const tree = renderer.create(
      <Flex alignStrech />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex aligned to the center', () => {
    const tree = renderer.create(
      <Flex alignCenter />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex aligned baseline', () => {
    const tree = renderer.create(
      <Flex alignBaseline />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex justified to the end', () => {
    const tree = renderer.create(
      <Flex justifyEnd />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex justified to the start', () => {
    const tree = renderer.create(
      <Flex justifyStart />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex justified strech', () => {
    const tree = renderer.create(
      <Flex justifyCenter />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex justified to the center', () => {
    const tree = renderer.create(
      <Flex justifyAround />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex justified baseline', () => {
    const tree = renderer.create(
      <Flex justifyBetween />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex aligned and justified', () => {
    const tree = renderer.create(
      <Flex justifyCenter alignCenter />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
