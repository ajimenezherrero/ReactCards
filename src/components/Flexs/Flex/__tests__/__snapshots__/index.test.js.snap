// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> renders div with flex aligned and justified 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex aligned baseline 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex aligned strech 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: strech;
  -webkit-box-align: strech;
  -ms-flex-align: strech;
  align-items: strech;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex aligned to the center 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex aligned to the end 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex aligned to the start 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex column direction 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex justified baseline 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex justified strech 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex justified to the center 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex justified to the end 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with flex justified to the start 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  className="c0"
/>
`;

exports[`<Flex /> renders div with inline flex 1`] = `
.c0 {
  width: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

<div
  className="c0"
/>
`;
